src/main/java/org/elasticsearch/index/query/IdsQueryParser.java

```diff
@@ -107,26 +98,13 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                 }
             }
         }
-
-        if (!idsProvided) {
-            throw new QueryParsingException(parseContext, "[ids] query, no ids values provided");
-        }
-
         if (ids.isEmpty()) {
-            return Queries.newMatchNoDocsQuery();
-        }
-
-        if (types == null || types.isEmpty()) {
-            types = parseContext.queryTypes();
-        } else if (types.size() == 1 && Iterables.getFirst(types, null).equals("_all")) {
-            types = parseContext.mapperService().types();
-        }
-
-        TermsQuery query = new TermsQuery(UidFieldMapper.NAME, Uid.createTypeUids(types, ids));
-        query.setBoost(boost);
-        if (queryName != null) {
-            parseContext.addNamedQuery(queryName, query);
+            throw new QueryParsingException(parseContext, "[ids] query, no ids values provided");
```

Comment:
This one is weird in the code before the refactoring:
```
if (!idsProvided) {
   throw new QueryParsingException(parseContext, "[ids] query, no ids values provided");
}
if (ids.isEmpty()) {
    return Queries.newMatchNoDocsQuery();
}
```
I wonder if this distiction makes sense, seems to be okay at the moment to provide no ids as longs as "values" field is set in the query, even if its empty.